addons/sale_management/migrations/12.0.1.0/pre-migration.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2019 Eficent <http://www.eficent.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+_column_renames = {
+    'sale_quote_template': [
+        ('require_payment', None),
+    ],
+    'sale_quote_option': [
+        ('layout_category_id', None),  # just in case, we save unused value
+    ],
+}
+
+_field_renames = [
+    ('sale.order', 'sale_order', 'template_id', 'sale_order_template_id'),
+    ('sale.quote.line', 'sale_quote_line', 'quote_id',
+     'sale_order_template_id'),
+    ('sale.quote.option', 'sale_quote_option', 'template_id',
+     'sale_order_template_id'),
+]
+
+_model_renames = [
+    ('sale.quote.line', 'sale.order.template.line'),
+    ('sale.quote.option', 'sale.order.template.option'),
+    ('sale.quote.template', 'sale.order.template'),
+]
+
+_table_renames = [
+    ('sale_quote_line', 'sale_order_template_line'),
+    ('sale_quote_option', 'sale_order_template_option'),
+    ('sale_quote_template', 'sale_order_template'),
+]
```

Comment:
the sale.order.option, sale.quote.template, sale.quote.line and sale.quote.option models are defined in the v11 website_quote module which may not be installed. In v12, the website_quote is renamed sale_quotation_builder.

In any case none of these two modules may be installed and then these renames just fail.